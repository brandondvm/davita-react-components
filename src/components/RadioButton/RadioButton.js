import React from 'react';
import PropTypes from 'prop-types';
import Input from '../Input/Input';
import idx from 'idx';

const RadioButton = ({ inputLabel, inputName, inputId, inputValue, isInputChecked, onOptionClick, textInput, radioText, errors = [], register = () => {}, validation, watchFields = [], watch = () => {} }) => {
  const error = idx(errors, _ => _[textInput.name]);
  return(
    <div className="radiobutton-component">
      <input
        type="radio"
        id={inputId}
        aria-checked={isInputChecked}
        defaultChecked={isInputChecked}
        value={inputValue}        
        name={inputName}
        aria-label={inputLabel}
        ref={register(validation)}
        className="visually-hidden"
        onChange={onOptionClick}
      />
      <label
        htmlFor={inputId}
        className={`radiobutton-component_label ${
          isInputChecked ? 'is-checked' : ''
        }`}
      >
        {inputLabel}
      </label>
      {watchFields[textInput?.watching] === inputValue && (
        <Input
          id={textInput.id}
          placeholder={textInput.placeholder}
          name={textInput.name}
          innerLabel={textInput.innerLabel}
          type={textInput.type}
          register={register(textInput.validation)}
          errorMessage={error?.type ? error.message : null}
          defaultValue={watch(textInput.name)}
        />
      )}
      {watchFields[radioText?.watching] === inputValue && (
        <p className="radiobutton-component_text">{ radioText.text }</p>
      )}
    </div>
  );
};

RadioButton.propTypes = {
  inputLabel: PropTypes.string,
  inputName: PropTypes.string,
  inputId: PropTypes.string,
  inputValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  isInputChecked: PropTypes.bool,
  onInputChange: PropTypes.func,
  textInput: PropTypes.object,
  register: PropTypes.func,
};

export default RadioButton;
