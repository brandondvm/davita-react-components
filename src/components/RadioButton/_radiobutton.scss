.radiobutton-component {
  display: flex;
  flex-flow: column;
  margin-bottom: 16px;
  width: 100%;

  input[type="radio"] {
    margin: 0;
	}

  &_name {
    color: $onyx;
    font-family: $regular-font;
    font-size: 14px;
    line-height: 17px;
  } 

	&_label {
    cursor: pointer;
		display: flex;

    &::before {
      content: "";
      display: block;
      background-image: url('../../images/radioButton.svg');
      background-position: center;
      background-repeat: no-repeat;
      background-size: 20px 20px;
      height: 20px;
      width: 20px; 
      margin-right: 10px;
    }   

		&.is-checked {
      &::before {
        background-image: url('../../images/radioButtonActive.svg');
      }
		}
  }

  &_text {
    color: $onyx;
    font-size: 14px;
    line-height: 17px;
    margin-bottom: 0;
    margin-top: 15px;
    span {      
      font-family: $bold-font;
    }
  }

  .input-component {
    margin-top: 16px;
  }
}
