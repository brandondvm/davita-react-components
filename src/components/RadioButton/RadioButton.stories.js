import React from 'react';

import RadioButton from './RadioButton';

export default {
  title: 'RadioButton',
  component: RadioButton,
  argTypes: {
    label: 'string'
  },
};

const Template = (args) => <div className="storybook-wrapper"><RadioButton {...args} /></div>;

export const Default = Template.bind({});
Default.args = {
  inputId: 'radio1',
  inputName: 'radio1',
  inputLabel: 'Radio Button',
  inputValue: 1,
  isInputChecked: false 
};
