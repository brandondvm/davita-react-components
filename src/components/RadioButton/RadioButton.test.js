import enzyme from 'enzyme';
import React from 'react';

import RadioButton from './RadioButton';

const mockProps = {
  inputLabel: 'Option 1', 
  inputName: 'radioName', 
  inputId: 'Option1',  
  inputValue: 1, 
  isInputChecked: true,
  watchFields:[],
  onOptionClick: jest.fn(),
  register: jest.fn(),
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<RadioButton { ...updatedProps } />);
}

describe('RadioButton Component', () => {
  test('Component renders correctly', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component should render props correctly on DOM', () => {
    const wrapper = setup();
    const inputEl = wrapper.find('input');
    const inputLabelEl = wrapper.find('.radiobutton-component_label');

    expect(inputEl.prop('value')).toEqual(mockProps.inputValue);
    expect(inputEl.prop('name')).toEqual(mockProps.inputName);
    expect(inputEl.prop('id')).toEqual(mockProps.inputId);
    expect(inputLabelEl.text()).toEqual(mockProps.inputLabel);
  });

  test('Component should render correctly input checked', () => {
    const wrapper = setup();
    const inputLabelEl = wrapper.find('.radiobutton-component_label');
    expect(inputLabelEl.hasClass('is-checked')).toBeTruthy();
  });

  test('Component should render correctly input unchecked', () => {
    const wrapper = setup({ isInputChecked: false });
    const inputLabelEl = wrapper.find('.radiobutton-component_label');
    expect(inputLabelEl.hasClass('is-checked')).toBeFalsy();
  });
});
