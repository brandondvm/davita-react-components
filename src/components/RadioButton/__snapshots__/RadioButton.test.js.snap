// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component Component renders correctly 1`] = `
<div
  className="radiobutton-component"
>
  <input
    aria-checked={true}
    aria-label="Option 1"
    className="visually-hidden"
    defaultChecked={true}
    id="Option1"
    name="radioName"
    onChange={[MockFunction]}
    type="radio"
    value={1}
  />
  <label
    className="radiobutton-component_label is-checked"
    htmlFor="Option1"
  >
    Option 1
  </label>
</div>
`;
